src/compiler.ts

```diff
@@ -6611,7 +6611,9 @@ export class Compiler extends DiagnosticEmitter {
         if (!overrideInstance.is(CommonFlags.Compiled)) continue; // errored
         let overrideType = overrideInstance.type;
         let originalType = instance.type;
-        if (!overrideType.isAssignableTo(originalType)) {
+
+        assert(originalType.getSignature() != null && overrideType.getSignature() != null);
+        if (!(overrideType.getSignature() as Signature).isAssignableTo(originalType.getSignature() as Signature, true)) {
```

Comment:
I will in general follow this comment. Use a `const overrideTypeSignature=` and  change `overrideSignature.toString(), originalSignature.toString()`

A small difference is:
`originalType` doesn't have `.signature`, it only has a `signatureReference: Signature | null`. Since there is a getter function of signature already, we should use the getter function instead of the under layer reference.

Reply:
I'm referring to `instance.signature` and `overrideInstance.signature` and not anything on `originalType` and `overrideType`.

Reply:
Ahh, understand, I will do it